### PR TITLE
Fix bug in non-interactive s.remove_background(fast=True)

### PR DESCRIPTION
**Pull request by magnunor**
Fixes bug in `s.remove_background(signal_range=(..., ...), fast=True)`, where the metadata and axes_manager information was not copied over to the result.

More info: https://github.com/hyperspy/hyperspy/issues/2104

- [x] Fix bug
- [ ] Add tests
- [ ] ready for review.

### CODE REVIEW

**Comment by magnunor** [comment]
@ericpre mentioned that this fix might lead to a slowdown when using `s.remove_background` interactively. Having compared this pull request with the current RELEASE_next_patch, there doesn't seem to be any difference. However, this might be due to my specific computer setup.

Could some else see if this fix reduces the responsiveness when running `s.remove_background` interactively?

File: hyperspy/_signals/signal1d.py [hunk 1]
```diff
@@ -1036,8 +1036,8 @@ def _remove_background_cli(
         if fast and not self._lazy:
             try:
                 axis = self.axes_manager.signal_axes[0].axis
-                result = Signal1D(self.data -
-                                  background_estimator.function_nd(axis))
+                result = self - Signal1D(
+                        background_estimator.function_nd(axis))
```
Comment (ericpre) [inline hyperspy/_signals/signal1d.py]: Actually, `Signal1D` is not necessary here and by removing it we can avoid the overhead of the creation of a signal, which make it 10% faster, not very significant but still something.

**Review by ericpre — CHANGES_REQUESTED** [review]
Actually, I was mistaken with the slow down in the [power law estimation](https://github.com/ericpre/hyperspy/blob/c314c8266c89dcacc04b9fe6f87668376e8abdd7/hyperspy/_components/power_law.py#L128) as discussed in #2032. The slow down here is only of ~10%.

File: hyperspy/_signals/signal1d.py (hunk 1 above)
Comment (magnunor) [reply hyperspy/_signals/signal1d.py]: Good point! Fixed.

**Comment by magnunor** [comment]
I added a bunch of tests to reduce the odds of `s.remove_background` silently breaking again.

**Comment by magnunor** [comment]
I also think I discovered some new bugs through these tests:

```python
s = hs.signals.Signal1D(np.arange(10, 100)[::-1])
s_r = s.remove_background(signal_range=(20, 50), background_type='Polynomial', fast=True)
print(s.data.shape)
(90,)
print(s_r.data.shape)
(3, 90)
```

Similar for `background_type='Offset'` gives `(90,)` and `(1, 90)`.

**Comment by magnunor** [comment]
Closing this, as it has been superseded by https://github.com/hyperspy/hyperspy/pull/2156

**Pull request closed without merging**